src/Uno.Toolkit.RuntimeTests/Tests/TabBarItemExtensionTests.cs

```diff
@@ -0,0 +1,221 @@
+
+		TabBarItem[] source = [tabBarItem];
+
+		var tabBar = CreateTabBar(source);
+
+		Grid.SetRow(frame, 0);
+		Grid.SetRow(tabBar, 1);
+
+		root.Children.Add(frame);
+		root.Children.Add(tabBar);
+
+		await UnitTestUIContentHelperEx.SetContentAndWait(root);
+
+		NavigateToLastPage(frame);
+
+		// Ensure it navigated to the last page
+		Assert.AreEqual(typeof(NavBarSimplePage), frame.Content.GetType());
+
+		tabBarItem.ExecuteTap();
+
+		Assert.AreEqual(typeof(NavBarSecondPage), frame.Content.GetType());
+	}
+
+	[TestMethod]
+	public async Task When_TabBarItem_ScrollToTop_ScrollViewer()
+	{
+		var root = CreateGrid();
+
+		var scrollViewer = new ScrollViewer();
+
+		// Create some content for the ScrollViewer (StackPanel with buttons)
+		var stackPanel = new StackPanel();
+		for (int i = 0; i < 50; i++)
+		{
+			stackPanel.Children.Add(new Button { Content = $"Button {i + 1}" });
+		}
+
+		scrollViewer.Content = stackPanel;
+
+		var tabBarItem = CreateTabBarItem(TBIOnClickBehaviors.ScrollToTop);
+
+		TabBarItem[] source = [tabBarItem];
+
+		var tabBar = CreateTabBar(source);
+
+		Grid.SetRow(scrollViewer, 0);
+		Grid.SetRow(tabBar, 1);
+		root.Children.Add(scrollViewer);
+		root.Children.Add(tabBar);
+
+		await UnitTestUIContentHelperEx.SetContentAndWait(root);
+
+		scrollViewer.ChangeView(0, 100, null, false);
+		await Task.Delay(_defaultDelayValue);
+		// Ensure scrolled down
+		Assert.AreEqual(100, scrollViewer.VerticalOffset);
+
+		tabBarItem.ExecuteTap();
+		await Task.Delay(_defaultDelayValue);
+		Assert.AreEqual(0, scrollViewer.VerticalOffset);
+	}
+
+#if __IOS__ || __ANDROID__ || WINDOWS_WINUI
+	[TestMethod]
+	public async Task When_TabBarItem_ScrollToTop_ListView()
+	{
+		var root = CreateGrid();
+
+		// Create a ListView with some items
+		var listView = new ListView();
+		listView.ItemsSource = Enumerable.Range(1, 50).Select(i => $"Item {i}");
+
+		// Optional: Add some styling to make the list more scrollable
+		listView.Height = 300;  // Set a fixed height for scrolling
+
+		var tabBarItem = CreateTabBarItem(TBIOnClickBehaviors.ScrollToTop);
+
+		TabBarItem[] source = [tabBarItem];
+
+		var tabBar = CreateTabBar(source);
+
+		Grid.SetRow(listView, 0);
+		Grid.SetRow(tabBar, 1);
+		root.Children.Add(listView);
+		root.Children.Add(tabBar);
+
+		await UnitTestUIContentHelperEx.SetContentAndWait(root);
+
+		var item50 = listView.Items[49];
+		listView.ScrollIntoView(item50);
+
+		await Task.Delay(_defaultDelayValue);
+
+		// Ensure the item is visible
+		var isItemVisible = IsItemVisible(listView, item50);
+		Assert.IsTrue(isItemVisible);
+
+		tabBarItem.ExecuteTap();
+
+		await Task.Delay(_defaultDelayValue);
+
+		isItemVisible = IsItemVisible(listView, item50);
+
+		Assert.IsFalse(isItemVisible);
+	}
+#endif
+
+	#region Helper methods
+	private static bool IsItemVisible(ListView list, object item)
+	{
+		var listViewItem = (ListViewItem)list.ContainerFromItem(item);
+		return listViewItem != null;
+	}
```

Comment:
I'm not sure if this is the best way to verify that it worked, although it functions correctly on Android, iOS, and Windows. Is there another way to check if the `ListView` scrolls to a specific position and then, after tapping the `TabBarItem`, it scrolls back to the top?

cc @kazo0 @Xiaoy312

Reply:
it only checks if the items is within materialization range (150% or 200% of viewport)
you should use TransformtoVisual to get the relative coords of LV vs its LVI:
```cs
item0.TransformToVisual(listView).TransformPoint(default).Y // check if == 0
```
note that you may want to compare against listview's itemspanel instead of listview, because padding/header/whatnot pushing the item down
or you could save the y-coord of the item0 before any scrolling, to compare against